lib/dx/error/error.ex

```diff
@@ -0,0 +1,7 @@
+defmodule Dx.Error do
+  defexception [:cause]
+
+  def message(error) do
+    "Error occurred: #{inspect(error.cause)}"
```

Comment:
I suggest adding `pretty: true` here, in case the `cause` is more complex.

```suggestion
    "Error occurred: #{inspect(error.cause, pretty: true)}"
```